src/controllers/manual-address.js

```diff
@@ -1,8 +1,105 @@
+import utils from 'naturescot-utils';
 import {ReturnState} from './_base.js';
 
-const manualAddressController = (_request) => {
-  // The only way out of the manual address page is onwards, so return success and begin
-  // the form.
+/**
+ * Clean the incoming POST request body to make it more compatible with the
+ * database and its validation rules.
+ *
+ * @param {any} body the incoming request's body
+ * @returns {any} a json object that's just got our cleaned up fields on it
+ */
+
+const cleanInput = (body) => {
+  return {
+    // The strings are trimmed for leading and trailing whitespace and then
+    // copied across if they're in the POST body or are set to undefined if
+    // they're missing.
+    addressLine1: body.addressLine1 === undefined ? undefined : body.addressLine1.trim(),
+    addressLine2: body.addressLine2 === undefined ? undefined : body.addressLine2.trim(),
+    addressTown: body.addressTown === undefined ? undefined : body.addressTown.trim(),
+    addressCounty: body.addressCounty === undefined ? undefined : body.addressCounty.trim(),
+    postcode: body.postcode === undefined ? undefined : body.postcode.trim()
+  };
+};
+
+const manualAddressController = (request) => {
+  // Clean up the user's input before we store it in the session.
+  const cleanForm = cleanInput(request.body);
+  request.session.addressLine1 = cleanForm.addressLine1;
+  request.session.addressLine2 = cleanForm.addressLine2;
+  request.session.addressTown = cleanForm.addressTown;
+  request.session.addressCounty = cleanForm.addressCounty;
+  request.session.postcode = cleanForm.postcode;
+
+  // Clear the errors
+  request.session.addressLine1Error = false;
+  request.session.townError = false;
+  request.session.countyError = false;
+  request.session.postcodeError = false;
+  request.session.invalidPostcodeError = false;
+
+  // Build the address array, ignoring any blank fields.
+  const address = [];
+  if (request.session.addressLine1 !== undefined && request.session.addressLine1.trim() !== '') {
+    address.push(request.session.addressLine1);
+  }
+
+  if (request.session.addressLine2 !== undefined && request.session.addressLine2.trim() !== '') {
+    address.push(request.session.addressLine2);
+  }
+
+  if (request.session.addressTown !== undefined && request.session.addressTown.trim() !== '') {
+    address.push(request.session.addressTown);
+  }
+
+  if (request.session.addressCounty !== undefined && request.session.addressCounty.trim() !== '') {
+    address.push(request.session.addressCounty);
+  }
+
+  if (request.session.postcode !== undefined && request.session.postcode.trim() !== '') {
+    address.push(request.session.postcode);
+  }
```

Comment:
I think that there's no need to build the address array if there are validation errors - so it would be better to put this code after the validation check and after the `ReturnState.Error` has been returned.